src/ast/codegen_llvm.cpp

```diff
@@ -232,9 +232,9 @@ void CodegenLLVM::visit(Builtin &builtin)
     }
     expr_ = b_.getInt64(builtin.probe_id);
   }
-  else if (builtin.ident == "args")
+  else if (builtin.ident == "args" || builtin.ident == "ctx")
   {
-    expr_ = ctx_;
+    expr_ = b_.CreatePtrToInt(ctx_, b_.getInt64Ty());
```

Comment:
`ctx_` is used:

- As an argument of helper functions
  - [CreateGetStackID()](https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/irbuilderbpf.cpp#L625), [CreatePerfEventOutput()](https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/irbuilderbpf.cpp#L667), [CreateOverrideReturn()](https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/irbuilderbpf.cpp#L696)
  - The function takes a Int8PtrTy value
- To read registers
  - ["arg"](https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/codegen_llvm.cpp#L512), ["sarg"](https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/codegen_llvm.cpp#L214), ["reg"](https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/codegen_llvm.cpp#L620), [CreateUSDTReadArgument()](https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/irbuilderbpf.cpp#L351)
  - the address is calculated by CreateGEP(), then a value is loaded
  - e.g., https://github.com/iovisor/bpftrace/blob/c532159263d5286ca1ca8120c85de48c1f39ac75/src/ast/codegen_llvm.cpp#L618-L621

The casting makes sense to me. It will make code more maintainable since bpftrace treats most variables as an integer. I'll work on it.

Reply:
I made the changes (https://github.com/mmisono/bpftrace/commit/86c657ccaa5773d7cecd397c8fff286ff2fa19ee), but this caused many codegen changes (18 test failed) and would conflict #1105. So I'll keep the initial code in this PR.